salt/_modules/metalk8s_blkid.py

```diff
@@ -0,0 +1,140 @@
+# coding: utf-8
+
+"""Wrapper exposing a Python-like interface around the libblkid functions.
+
+Why this module instead of shelling out to `blkid`?
+===================================================
+
+Because we want to be as sure as possible that when `blkid` returns no
+informationg is means "there is no info" and not "maybe there is some info but I
+couldn't get them" (because if we mix two cases, we may actually format
+something because we thought "ho there is nothing, davai" whereas the reality is
+"there was data but I haven't seen them because reasons…").
+
+Relying on `return code 2 == no data` seems kinda safe, but it's not 100% clear
+from the man page. Now we could check the source code (and I did) but that would
+only be valid for the current version we look at.
+
+So let's directly attack the library, making sure that any API call will result
+in an exception and thus if we end up with "no data" then that truly means
+"everything went well, but there was nothing there".
+"""
+
+import collections
+import ctypes
+import functools
+
+import metalk8s_blkid_c as blkid_c
```

Comment:
You can't do this in Salt. It may work, but it's not guaranteed to work.

If you have code that's 'utility code', i.e. not 'real' Salt-module functionality, have it as a 'utils' module and consume it from `__utils__`.

Although in this case I'd merge both files in one.

Reply:
I'll move those into `_utils` then.

I split the code between two files to separate the "low-level" binding that only defines the "interfacing" from the "high-level" binding that provides a Python-like interface on top of it.

But yeah, I can merge those, since I'll already using folds I can use them to keep the separate.

Done